ext/pcntl/pcntl.c

```diff
@@ -385,6 +403,41 @@ PHP_FUNCTION(pcntl_waitpid)
 }
 /* }}} */
 
+#ifdef HAVE_WAITID
+PHP_FUNCTION(pcntl_waitid)
+{
+	zend_long idtype = P_PID;
+	zend_long id = 0;
+	zval *user_siginfo = NULL;
+	zend_long options = WEXITED;
+	int success;
+
+	ZEND_PARSE_PARAMETERS_START(2, 4)
+		Z_PARAM_LONG(idtype)
+		Z_PARAM_LONG(id)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_ZVAL(user_siginfo)
+		Z_PARAM_LONG(options)
+	ZEND_PARSE_PARAMETERS_END();
+
+	errno = 0;
+	siginfo_t siginfo;
+
+	success = waitid((idtype_t) idtype, (id_t) id, &siginfo, (int) options);
```

Comment:
I would merge the declaration and assignment and rename the variable, as `success` is not very accurate.
```suggestion

	ZEND_PARSE_PARAMETERS_START(2, 4)
		Z_PARAM_LONG(idtype)
		Z_PARAM_LONG(id)
		Z_PARAM_OPTIONAL
		Z_PARAM_ZVAL(user_siginfo)
		Z_PARAM_LONG(options)
	ZEND_PARSE_PARAMETERS_END();

	errno = 0;
	siginfo_t siginfo;

	int status = waitid((idtype_t) idtype, (id_t) id, &siginfo, (int) options);
```